src/triage/risklist/__init__.py

```diff
@@ -0,0 +1,171 @@
+            feature_names_in_group = [f for f in model_info['feature_names'] if re.match(f'\A{feature_prefix}', f)]
+            reconstructed_feature_dictionary[feature_group] = feature_names_in_group
+
+            # Make sure that the features imputed in training should also be imputed in production
+            features_imputed_in_train = [f for f in set(feature_names_in_group) if f + '_imp' in feature_names_in_group]
+            results = conn.execute(aggregation.find_nulls())
+            null_counts = results.first().items()
+
+            features_imputed_in_production = [col for (col, val) in null_counts if val > 0]
+            total_impute_cols = set(features_imputed_in_production) | set(features_imputed_in_train)
+            total_nonimpute_cols = set(f for f in set(feature_names_in_group) if '_imp' not in f) - total_impute_cols
+            task_generator = feature_generator._generate_imp_table_tasks_for
+            imputation_table_tasks.update(task_generator(
+                aggregation,
+                impute_cols=list(total_impute_cols),
+                nonimpute_cols=list(total_nonimpute_cols)
+                )
+            )
+    feature_generator.process_table_tasks(imputation_table_tasks)
+
+    # 5. Build matrix
+    db_config = {
+        "features_schema_name": "production",
+        "labels_schema_name": "public",
+        "cohort_table_name": cohort_table_name,
+    }
+
+    matrix_builder = MatrixBuilder(
+        db_config=db_config,
+        matrix_storage_engine=matrix_storage_engine,
+        engine=db_engine,
+        experiment_hash=None,
+        replace=True,
+    )
+
+    matrix_metadata = {
+        'as_of_times': [as_of_date],
+        'matrix_id': str(as_of_date) + '_prediction',
+        'state': DEFAULT_ACTIVE_STATE,
+        'test_duration': '1y',
+        'matrix_type': 'production',
+        'label_timespan': None,
+        'label_name': model_info['label_config']['name'],
+        'indices': ["entity_id", "as_of_date"],
+        'feature_start_time': model_info['feature_start_time'],
+    }
+
+    matrix_uuid = filename_friendly_hash(matrix_metadata)
+
+    matrix_builder.build_matrix(
+        as_of_times=[as_of_date],
+        label_name=model_info['label_config']['name'],
+        label_type=None,
+        feature_dictionary=reconstructed_feature_dictionary,
+        matrix_metadata=matrix_metadata,
+        matrix_uuid=matrix_uuid,
+        matrix_type="production",
+    )
+
+    # 6. Predict the risk score for production
+    predictor = Predictor(
+        model_storage_engine=model_storage_engine,
+        db_engine=db_engine
+    )
+
+    predictor.predict(
+        model_id=model_id,
+        matrix_store=matrix_storage_engine.get_store(matrix_uuid),
+        misc_db_parameters={},
+        train_matrix_columns=matrix_storage_engine.get_store(model_info['original_matrix_uuid']).columns()
```

Comment:
We may be keeping both train and test in memory here, whereas if we moved the train columns calculation to its own line that may not happen.